src/database.rs

```diff
@@ -103,7 +103,7 @@ impl Database {
         self.build_keydir(&mut data_files_sorted)
             .map_err(|source| Error::KeyDirFill {
                 path: base_dir.to_path_buf(),
-                source: source,
```

Comment:
if the field initialization has the same name as the variable, then you don't need to repeat it.